app/routes/aoi.py

```diff
@@ -125,16 +131,17 @@ async def get_aoi_by_bbox(
         80,
         description="Minimum probability threshold for plastic detection (0-100)",
     ),
+    db: Session = Depends(get_db),
```

Comment:
dependency injection pattern? :)  Or does the db as parameter has a different reason?

Reply:
Whats the benefit of using fastAPI "Depends"

Reply:
mainly dependency injection yes. Also to abstract the try except for session closing. this way we can reuse it and test the routes more easily